test/kubernetes_test.go

```diff
@@ -351,16 +696,40 @@ func doIntegrationTests(t *testing.T, corefile string, testCases []test.Case) {
 			t.Fatalf("Could not send query: %s", err)
 		}
 
-		// check the answer
-		if res.Rcode != tc.Rcode {
-			t.Errorf("Expected rcode %d but got %d for query %s, %d", tc.Rcode, res.Rcode, tc.Qname, tc.Qtype)
+		// Before sorting, make sure that CNAMES do not appear after their target records
```

Comment:
This CNAME answer order check can be made into a function and put in middleware/test/helpers.go. e.g. test.CNAMEOrder(res.Answer).